### PR TITLE
SW-1603 Show Species Database Error to User

### PR DESCRIPTION
**Pull request by charlie-tf**
- Add snackbar to problem tooltip
- Get the error message from the database call
- Display in an error toast if exists
image

### CODE REVIEW

File: src/api/species/species.ts [hunk 1]
```diff
@@ -97,6 +97,7 @@ type SimpleSuccessResponsePayload =
  */
 export type UpdateSpeciesResponse = {
   requestSucceeded: boolean;
+  responseMessage?: string;
```
Comment (karthikbtf) [inline src/api/species/species.ts:100]: this is fine... here's an example of how we have been naming the property recently 
https://github.com/terraware/terraware-web/blob/main/src/api/accessions2/accession.ts#L27
only reason we added a util is to remove some boilerplate code

File: src/components/Species/ProblemTooltip.tsx [hunk 2]
```diff
@@ -62,7 +64,10 @@ export default function ProblemTooltip({
   };
 
   const acceptFix = async (problemId: number) => {
-    await acceptProblemSuggestion(problemId);
+    const response = await acceptProblemSuggestion(problemId);
+    if (!(response && response.requestSucceeded)) {
```
Comment (karthikbtf) [inline src/components/Species/ProblemTooltip.tsx:68]: can just be `if (!response.requestSucceeded) ` since by contract you will get a response

**Review by karthikbtf — APPROVED** [review]
some suggestions

File: src/api/species/species.ts [hunk 3]
```diff
@@ -276,11 +277,10 @@ export async function acceptProblemSuggestion(problemId: number) {
   const response: UpdateSpeciesResponse = {
     requestSucceeded: true,
   };
-  try {
-    await axios.post(SPECIES_PROBLEM.replace('{problemId}', problemId.toString()));
-  } catch {
+  await axios.post(SPECIES_PROBLEM.replace('{problemId}', problemId.toString())).catch((e) => {
```
Comment (karthikbtf) [inline src/api/species/species.ts:280]: Not sure why I missed this.. Try/catch would be the better pattern with await.. curious why the promise like handling.

Comment (karthikbtf) [reply src/api/species/species.ts:280]: one is an asynchronous pattern
```
  return promise.then().catch();
```
the other is a blocking pattern
```
  try {
     return await promise;
  } catch(e) {
     return error
  }
```
there is no hard and fast rule... its mostly for readability.
you can mix/match them for sure but no point in using await..
plenty of references online .. hate pointing to links from random sources but we all like stackoverflow.. so here you go https://stackoverflow.com/questions/69362121/when-should-i-use-try-catch-instead-of-then-catch
once again, its a matter of readability versus theoretical correctness.

Comment (karthikbtf) [reply src/api/species/species.ts:280]: fwiw  - before introduction of `await` a lot of us were in promise chain hell.. hard to read/follow/debug code...

Comment (charlie-tf) [reply src/api/species/species.ts:280]: Thank you. I will make the change ❤️

**Review by karthikbtf — APPROVED** [review]
awesome, thanks!

**Pull request merged**